File: leetcode/easy/two-sum/two-sum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> indices = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            if (indices.get(target - nums[i]) != null) {
                return new int[] {i, indices.get(target - nums[i])};
            }
            indices.put(nums[i], i);
        }
        return new int[] {};
    };
}

File: leetcode/easy/two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  const indices = {};

  for (let i = 0; i < nums.length; i++) {
      let num = nums[i];
      if (indices[target - num] !== undefined) 
          return [indices[target - num], i]
      indices[num] = i;
  }
};